### PR TITLE
Restore release cadence

### LINKED ISSUE #136: Unable to use module with bridge network_mode and without load balancers.
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help. 

[Slack Community](https://slack.cloudposse.com)

## Describe the Bug
Looks like it is not possible to create service without defining ecs_load_balancers variable and use "bridge" network_mode.
We receive a error:
module.ecs_alb_service_task.aws_ecs_service.default[0]: Creating...
╷
│ Error: error creating filebeat service: error waiting for ECS service (filebeat) creation: InvalidParameterException: IAM roles are only valid for services configured to use load balancers.

Look like local variable  "enable_ecs_service_role" has wrong conditions.
enable_ecs_service_role = module.this.enabled && var.network_mode != "awsvpc" && length(var.ecs_load_balancers) <= 1

It should be: length(var.ecs_load_balancers) >= 1

## Expected Behavior
Able to create service in Bridge mode without load balancers.

## Environment (please complete the following information):

Terraform v1.0.5
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.61.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/template v2.2.0

### LINKED ISSUE #139: A managed resource "aws_security_group" "ecs_service" has not been declared in module.ecs_alb_service_task.
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help. 

[Slack Community](https://slack.cloudposse.com)

## Describe the Bug
```
  on .terraform/modules/ecs_alb_service_task/main.tf line 435, in resource "aws_ecs_service" "ignore_changes_task_definition_and_desired_count":
 435:       security_groups  = compact(concat(var.security_group_ids, aws_security_group.ecs_service.*.id))

A managed resource "aws_security_group" "ecs_service" has not been declared in
module.ecs_alb_service_task.

Releasing state lock. This may take a few moments...
```

## Suggested change

https://github.com/cloudposse/terraform-aws-ecs-alb-service-task/pull/140 PR here.

`security_groups  = compact(concat(var.security_group_ids, aws_security_group.ecs_service.*.id))`
find and replace with
`security_groups = compact(concat(module.security_group.*.id, var.security_groups))`

## Why!!!?
Looks like there's a bug in the code if you use the `ignore` `desired_count` and `task_definition` flags! 😨 

Setting `var.ignore_changes_task_definition` or `var.ignore_changes_desired_count` or both(?) to `true` results in and error:

```
A managed resource "aws_security_group" "ecs_service" has not been declared in
module.ecs_alb_service_task.
```
I think that error makes sense as we don't have a `aws_security_group.ecs_service` resource in the module.

I took at look at the code [further down at](https://github.com/cloudposse/terraform-aws-ecs-alb-service-task/blob/0922981957cfd72aa80c363c263cf54394a518f2/main.tf#L620) (the `default`)

and that (and this is what the tests use) references the `module` `security_group`: module.security_group.*.id

Which does exist 👍

### LINKED ISSUE #142: Expose uniqueid of task_exec role
## Describe the Feature

An output of the unique IDs for the two roles created for the service task:
task_exec_role_uniqueid

## Expected Behavior

When using this module, I can access the uniqueid (https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role#unique_id) field on the task_exec role.

## Use Case

We use the unique ids for permissions for secrets. Since I cannot access the uniqueid on the task_exec role, we have to do a chicken and an egg scenario of trying to access these fields through a data source:

```
# These are needed because the cloudposse module does not expose the roles unique ids
data "aws_iam_role" "task_exec_role" {
  name = module.ecs_alb_service_task.task_exec_role_name
}
```

## Describe Ideal Solution

As stated above, just an output of the uniqueid for the task_exec role

### PR DESCRIPTION
**Pull request by Nuru**
## what and why

- Reverts #117, #120, #151 to restore the ability to move forward on bug fixes and new features while maintaining backward compatibility
- Supersedes and closes #118, closes #136, via #137, #145, thanks to @ragumix, @verbalius, @asiragusa
- Reimplements #120 reverted above to maintain that functionality, closes #122, thanks to @flywheelnz, @joe-niland 
- Fixes #123
- Supersedes #125, thanks to @nitrocode
- Supersedes and closes #129, closes #130, closes #133
- Supersedes and closes #138, thanks to @mrsufgi
- Closes #139, supersedes and closes #140, thanks to @fentonfentonfenton
- Supersedes #143, closes #142, thanks to @missylbytes
- Supersedes #144, thanks to @mrsufgi
- Supersedes, implements, and closes #151, closes #152, thanks to @ethanrubio 
- Supersedes #154, thanks to @korenyoni

### CODE REVIEW

**Comment by Nuru** [comment]
/test all

NOTE: Failure of Validate Codeowners is to due to restrictions placed on bots on this repository and is not significant.

**Review by nitrocode — APPROVED** [review]
LGTM

File: README.yaml [hunk 1]
```diff
@@ -229,5 +197,3 @@ contributors:
   github: nitrocode
 - name: Maxim Mironenko
   github: maximmi
-- name: Vladimir Syromyatnikov
-  github: SweetOps
```
Comment (korenyoni) [inline README.yaml:233]: I believe this comes from a git revert, but I would leave this in

**Review by korenyoni — APPROVED** [review]
LGTM but left a comment

**Pull request merged**